### ISSUE TITLE
Support for code block wrapped with "~~~"

### ISSUE THREAD

**Issue by Masynchin** [microsoft/pylance-release#4612]
Type: <b>Feature Request</b>

In my python function's docstring I wrote a code block, wrapped with `~~~`. Turns out, it doesn't recognized as code block, as opposed to ` ``` ` variant. I want to use both for code blocks.

Extension version: 2023.7.20
VS Code version: Code 1.80.0 (Universal) (660393deaaa6d1996740ff4880f1bad43768c814, 2023-07-04T13:39:33.766Z)
OS version: Darwin arm64 22.2.0
Modes:

**Comment by rchiodo** [comment]
Thanks for the issue but I'm not sure I understand what you're asking for. Can you describe a more concrete example?

Is it something like this?

```python
def foo():
    '''
    Doc string goes here
    ~~~
    Code block goes here?
    ~~~
    '''
```

If that's what you're looking for I believe we turn all doc strings into markdown. Markdown usually follows this format:

```python
def foo():
   """
   Comment here:
   ```python
   def bar():
      pass
   ```
   """
```

![image](https://github.com/microsoft/pylance-release/assets/19672699/3793ad78-8dff-4e94-a6a7-e450d686d0a6)

**Comment by rchiodo** [comment]
Oh maybe github does :)

~~~python
def foo():
   pass
~~~

**Comment by rchiodo** [comment]
Typescript in VS code does too. 

![image](https://github.com/microsoft/pylance-release/assets/19672699/db3bf7e5-9e93-47db-b1ec-6601f9326e01)

**Comment by rchiodo** [comment]
Okay seems like we should just support this as it seems to be a standard for other tools and languages.

**Comment by Masynchin** [comment]
Yes, this is what I meant. I should have added this example:

~~~python
def example():
    """
    Wrapped with backticks:

    ```python
    def function(a, b): ...
    ```

    Wrapped with tildas:

    ~~~python
    def function(a, b): ...
    ~~~
    """
~~~

<img width="256" alt="Rendered signature of the function above" src="https://github.com/microsoft/pylance-release/assets/47028153/6fc6f1d0-024d-49ce-bd76-f588cea457fb">

Here I want that both "backticks" and "tildas" versions were rendered as code block, but only first one is rendered correctly.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.7.43, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023743-31-July-2023-PreRelease)
